tests/test_rf_base.py

```diff
@@ -231,6 +231,32 @@ def _forward_step(*, model: _Net, extern_data: TensorDict):
     run_model(extern_data, lambda *, epoch, step: _Net(), _forward_step)
 
 
+def test_linear():
+    time_dim = Dim(Tensor("time", [batch_dim], dtype="int32"))
+    in_dim, out_dim = Dim(7, name="in"), Dim(13, name="out")
+    extern_data = TensorDict(
+        {
+            "data": Tensor("data", [batch_dim, time_dim, in_dim], dtype="float32"),
+            "classes": Tensor("classes", [batch_dim, time_dim], dtype="int32", sparse_dim=out_dim),
+        }
+    )
+
+    class _Net(rf.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = rf.LinearWithWeightDropout(in_dim, out_dim, weight_dropout_prob=0.1)
+
+        def __call__(self, x: Tensor) -> Tensor:
+            return self.linear(x)
+
+    # noinspection PyShadowingNames
+    def _forward_step(*, model: _Net, extern_data: TensorDict):
+        out = model(extern_data["data"])
+        out.mark_as_default_output()
+
+    run_model(extern_data, lambda *, epoch, step: _Net(), _forward_step)
```

Comment:
Also note, this test will actually not test at all the weight dropout. It will operate in forward mode, i.e. disable any dropout. But even if you enable training, what you also would want to check is whether it did the grad checkpointing actually correctly. Although I have no idea how to test this. In TF, I have similar tests by really investigating the computation graph. Not sure how you can investigate the gradient tape in PT. Or how you can verify in any way that the gradient checkpointing worked.